src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFunction = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      }
+
+      if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    let compare;
+
+    if (typeof compareFunction === 'function') {
+      compare = compareFunction;
+    } else {
+      compare = defaultCompareFunction;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm is correctly implemented, but it is not the most efficient sorting algorithm for larger datasets. If performance is a concern, consider researching and implementing more efficient algorithms like quicksort or mergesort.